docs: overhaul README structure

Bad3r opened the pull request:
## view the updated version
@https://github.com/logseq/logseq/blob/enhancement/overhaul-readme.md/README.md

## Changes:
- [x] Added logseq logo to the readme
- [x] Update the introduction format 
- [x] Update the types of badges used
- [x] change badges colors to fit Logseq colors 
- [x] General readability improvements 
- [x] emojis!
- [x] moved open collective user images to the thanks section
- [x] Based on feedback from discord; I also added a section to thank contributors
- [x] update logseq screenshot
- [x] Update logseq description
- [x] 3 screenshots, text, whiteboard, GTD
  - [x] turn into GIF

ens100 commented:
I think this looks a lot tidier and to the point than the current readme - just a few suggestions:

1 - Remove the numbers after "Download Logseq Desktop App" if possible - I am not it adds much and is a bit confusing of what it represents 
2 - Is the Product Hunt so important that it needs to be made so prominent? If not, would suggest to move it to the bottom of the page
3 - Lose the "collaboration" in the text or put "(coming soon)". 
4 - I would be inclined to change the text to match the website;:
"Logseq is a privacy-first, open-source knowledge base that works on top of local plain-text Markdown and Org-mode files." 

"Use it to write, organize and share your thoughts, keep your to-do list, and build your own digital garden."
5 - Is it worth having a link to the blog - although this might create too many links

Bad3r commented:
@ens100 great points!

1. The number represents the total downloads of logseq since the first release. I do see your point that its not clear. I can replace it with the version number or remove it and just leave the text
2. I don't really know; I just kept it in the same relevant position. Someone from the team or specifically @tiensonqin should decide that
3. yeah I didn't change the original but I agree that it should match the website but the new website :eyes: 
4. I was being lazy about creating a custom badge for the blog xD. I do agree it would be nice to have. I'll look into it

Bad3r commented:
This is ready for review

RCmerci commented:
The screenshot is outdated， how about replacing it with a new one？

hdansou commented:
A new screenshot:

![image](https://user-images.githubusercontent.com/536716/210489314-0d89f4ea-e36e-428f-b822-a532420ac7bc.png)

Bad3r commented:
@RCmerci screenshot update:  
 

![](https://user-images.githubusercontent.com/25513724/210489764-a50c2f89-28fc-4012-a5f2-311119ca7415.png)

tiensonqin commented:
@Bad3r The new readme and screenshot look beautiful!
> 2 - Is the Product Hunt so important that it needs to be made so prominent? If not, would suggest to move it to the bottom of the page

Let's move this to the bottom of the page.

>  Your data are plain text files and we currently support both Markdown and Emacs Org mode (more to be added soon)

There's no plan to support other formats than Markdown and Emacs Org mode now, please remove the part `more to be added soon`.

situ2001 commented on README.md:
```diff
@@ -1,99 +1,206 @@
+
+And here is a list of some [good first issues](https://github.com/logseq/logseq/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)!
+
+<br></br>
```
Non-blocking

Thanks for the amazing `README.md`! I found that there are some huge line breaks. In my opinion, the rendered heading1 (`#`) and heading2 (`##`) both have margins and a divider line, so the user can quickly figure out which section they are reading.

<img width="502" alt="image" src="https://user-images.githubusercontent.com/28241963/215058112-3b8350fd-ea79-4e9d-ba51-6ddfd57c1485.png">

Bad3r replied:
Yeah I agree, I'll change it

Bad3r commented:
> The screenshot is still not very attractive 🤔 

Yeah I was talking with Jackob (not sure what's his GH handle) about creating a render instead of a screenshot. 

He is also planning on making a banner for the top of the readme instead of the logo.

situ2001 commented on README.md:
```diff
@@ -1,65 +1,130 @@
+        <img src="https://img.shields.io/github/license/logseq/logseq?color=%2385c8c8&style=for-the-badge"/></a>
+    <a href="https://github.com/logseq/logseq/releases">
+        <img src="https://img.shields.io/github/v/release/logseq/logseq?color=%2385c8c8&style=for-the-badge"
+            alt="latest release version"></a>
```
I have found an interesting badge that indicates the download count of all releases.  I don't know whether to add it or not. The badge looks convincing.

```markdown
![GitHub all releases](https://img.shields.io/github/downloads/logseq/logseq/total?style=for-the-badge)
```

![GitHub all releases](https://img.shields.io/github/downloads/logseq/logseq/total?style=for-the-badge)

Bad3r replied:
I had it there. User feedback suggested that it's not necessary and confusing to average users. It's a cool stat, though!

logseq-cldwalker commented:
@Bad3r I can take a look when this is ready for another review

Bad3r commented:
@cldwalker, @cnrpman ready for review.

I updated the screenshot. We can replace it in the future with a render.

situ2001 commented on README.md:
```diff
@@ -1,96 +1,209 @@
+[Logseq](https://logseq.com) stores your notes locally on your device in plain text files, ensuring that your data is **private** and **secure**. Logseq also strongly emphasizes [**user control**](https://www.gnu.org/philosophy/free-sw.en.html), allowing you to access and manage your notes confidently.
 
-In the unlikely event that the website is down or cannot be maintained, your data is, and will always be yours.
+In addition to its core features, Logseq has a growing ecosystem of plugins and themes that enable a wide range of workflows and customization options. Mobile apps are also available, providing access to most features from the desktop application.
```
most features from the desktop application -> most of the features of the desktop application

Bad3r replied:
Both are grammatically correct. I prefer to be less wordy.

situ2001 commented on README.md:
```diff
@@ -1,96 +1,209 @@
+
+## 🫶 Support Logseq Development
+
+If you find Logseq useful and want to help us keep the project growing, please consider supporting our contributors on [Open Collective](https://opencollective.com/logseq). Your support is a way of showing our contributors that their efforts are appreciated and motivates them to continue their excellent work. Every contribution, no matter how small, helps us to keep improving Logseq.
```
helps us to keep -> helps us keep

situ2001 replied:
motivates them -> motivating them

Bad3r replied:
> helps us to keep -> helps us keep

Both are correct. Will update to the shorter version.

> motivates them -> motivating them

This is an incorrect suggestion

> Your support **shows** our contributors that their efforts are appreciated and **motivates** them to continue their excellent work.

**`shows`** `and` **`motivates`**

situ2001 commented on README.md:
```diff
@@ -1,96 +1,209 @@
+To start contributing to Logseq, please follow the guidelines outlined in our `CONTRIBUTING.md` file. We have a dedicated [overview page](CODEBASE_OVERVIEW.md) for Logseq's codebase and a [development practices guide](docs/dev-practices.md) that provides detailed information on our development process.
 
-## Learn more
+We welcome all contributions, big or small, and we appreciate your time and effort in helping us to improve Logseq for our users.
```
helping us to -> helping us

situ2001 commented on README.md:
```diff
@@ -1,96 +1,209 @@
+
+# 🙏 Thank You
+
+We want to express our sincere gratitude to our Open Collective sponsors, backers, and contributors. Your support and contributions allow us to continue developing and improving Logseq. Thank you for being a part of our community and helping us to make Logseq the best it can be!
```
Also `helping us`

cnrpman reviewed (approved):
Looks great!
Can we have a section & screenshot for the whiteboard feature?

situ2001 reviewed (commented):
Thanks for the README! I have read it throughly and caught some grammatical errors 🐛

Bad3r commented:
> Looks great! Can we have a section & screenshot for the whiteboard feature?

Added to the checklist. Will ping you once its ready.

Bad3r commented:
> Looks great! Can we have a section & screenshot for the whiteboard feature?

Not sure where to put the section. We do not mention any specific features in the readme. 
Should I add a section for major features, including whiteboards + screenshot? 🤔

situ2001 commented:
> Not sure where to put the section. We do not mention any specific features in the readme. Should I add a section for major features, including whiteboards + screenshot? 🤔

Here is my suggestion: 

## Main Features

Screenshot
Note taking (Centered Text)

Screenshot
Whiteboard (Centered Text)

Screenshot
More, like GTD, graph view, x-callback-url ,etc. (Centered Text)

Bad3r commented:
@cnrpman @situ2001 I updated the screenshots and the last revision of the doc.

@logseq-cldwalker, this is ready for review whenever you have the time.

Bad3r commented:
I incorporated @situ2001's feedback and added the suggested screenshots. @ens100 recommended adding a PDF annotation screenshot, so I created one and included it as well. However, the text in the PDF is not very clear at the window size used. To address this, I tried to find a book/page with wide spaces between the text to make the issue less severe.

Bad3r commented:
Any thoughts on using a GIF instead of 3 photos?
![logseq-demo](https://user-images.githubusercontent.com/25513724/221372974-e7823df2-6b0e-495b-ade9-cbc4f2cb842a.gif)
![grey-text-demo](https://user-images.githubusercontent.com/25513724/221373005-f45beb7e-2b16-490d-be48-042d2037d834.png)
![grey-whiteboard-demo](https://user-images.githubusercontent.com/25513724/221373006-7b4932ae-0029-4cfc-a2bc-f204ed626272.png)
![adelaide-GRD-demo](https://user-images.githubusercontent.com/25513724/221373007-5cb1e5e1-4626-4adb-ba07-4e4bea07aec6.png)

hdansou commented:
I have reorganized and rephrased the content a bit. You can read it here https://github.com/hdansou/logseq/tree/readme-update#readme.

Bad3r commented:
@hdansou thank you. Some feedback:  
1\. the header is too crowded and does not work as expected on mobile devices.  
   there is also a plan to replace the logo with a banner in the future.  
2\. having a table of content makes it easier for users looking for developer resources to find it, as it's most likely one of the most common reasons to read the readme

3\. I am not a fan of having the screenshot above text that would be a better fit for the upcoming banner 

4\. I do not like to use words like 'new feature'. It's unlikely that this potion of the readme will be updated anytime soon, and it can make the document outdated faster.

Over all it is good, but I prefer the current layout.

andelf commented:
The new README is really nice.

**non-blocking** For better Visual Identity:

- ![logo with original color](https://img.shields.io/twitter/follow/logseq?label=%40logseq&logo=twitter&style=for-the-badge) default
- ![logo with logseq theme](https://img.shields.io/twitter/follow/logseq?color=%2385c8c8&label=%40logseq&logo=twitter&style=for-the-badge) logo color untouched
- ![logo alternative](https://img.shields.io/twitter/follow/logseq?color=%2385c8c8&label=%40logseq&logo=twitter&logoColor=%2385c8c8&style=for-the-badge) current PR

Personally, I prefer the second.

🤣 I'll need some feedback from UX experts.

Bad3r commented:
Yeah its probably better to stick to logo colors. @sprocketc, @bendyorke any thoughts or feedback on the readme?

andelf commented:
@Bad3r Let's use the second logo style for better Visual Identity.

Bad3r commented:
> @Bad3r Let's use the second logo style for better Visual Identity. 

Sounds good will update it now.

Bad3r commented:
@andelf done!
@tiensonqin please take a look to confirm before merging the updated readme.
https://github.com/logseq/logseq/blob/enhancement/overhaul-readme.md/README.md

andelf reviewed (approved):
LGTM.

Bad3r commented:
@andelf thank you! any idea why the test is not running? I have been trying to trigger it without luck

logseq-cldwalker reviewed (approved):
@Bad3r Thanks for doing a readme makeover! 👍 ❤️ I think some of the contributing section can be moved so tweaking that in a followup commit. Not worried about tests as the only test that could fail is the typos one and I've confirmed that locally

Bad3r commented:
> @Bad3r Thanks for doing a readme makeover! 👍 ❤️ I think some of the contributing section can be moved so tweaking that in a followup commit. Not worried about tests as the only test that could fail is the typos one and I've confirmed that locally

Awesome thank you. Glad it's finally done and I think it's a great change from the old one.

RCmerci commented:
Hi @Bad3r, I just noticed that the num of sponsors isn't correct, it should be 300+ according to the opencollective page
https://opencollective.com/logseq

Bad3r commented:
@RCmerci yeah, I see. It gets pulled from:
https://shields.io/category/funding

Not sure why they are getting the wrong number. Maybe an issue with OC API
